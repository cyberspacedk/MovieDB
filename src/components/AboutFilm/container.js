import { connect } from 'react-redux';
import AboutFilm from './component';
import {
  isError,
  isLoading,
  getImages,
  getCasts,
  getCrew,
  getGenres,
  getFilmInfo,
} from '../../store/singleFilm/selectors';

const mstp = state => ({
  isError: isError(state),
  isLoading: isLoading(state),
  aboutFilm: getFilmInfo(state),
  images: getImages(state),
  casts: getCasts(state),
  crew: getCrew(state),
  genres: getGenres(state),
});
export default connect(mstp)(AboutFilm);
