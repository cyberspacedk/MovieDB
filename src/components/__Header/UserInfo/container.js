import { connect } from 'react-redux';
import UserInfo from './component';
import { userLogout } from '../../../store/authentifiction/actions';
import {
  getUserLogin,
  isAuthentificated,
} from '../../../store/authentifiction/selectors';

export const mstp = state => ({
  username: getUserLogin(state),
  sessionID: isAuthentificated(state),
});

const mdtp = {
  userLogout,
};

export default connect(
  mstp,
  mdtp,
)(UserInfo);
