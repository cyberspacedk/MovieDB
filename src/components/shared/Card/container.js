import React from 'react';
import PropTypes from 'prop-types';
import { Modal } from 'antd';
import Card from './component';

class CardItemContainer extends React.Component {
  removeMovieModal = e => {
    e.stopPropagation();

    const { history, item, operations } = this.props;

    Modal.confirm({
      title: 'Do you want to delete movie ?',
      onOk() {
        if (history.location.pathname.includes('lists')) {
          const listId = /[0-9]{2,}$/.exec(history.location.pathname)[0];
          operations(listId, item.id);
        } else {
          operations(item.id, false);
        }
      },
      onCancel() {},
    });
  };

  showMoreDetails = () => {
    this.props.history.push(`/${this.props.item.id}`);
  };

  render() {
    return (
      <Card
        showMoreDetails={this.showMoreDetails}
        removeMovieModal={this.removeMovieModal}
        {...this.props}
      />
    );
  }
}

CardItemContainer.propTypes = {
  item: PropTypes.shape(PropTypes.string),
  history: PropTypes.shape(PropTypes.object),
  operations: PropTypes.func,
};

export default CardItemContainer;
