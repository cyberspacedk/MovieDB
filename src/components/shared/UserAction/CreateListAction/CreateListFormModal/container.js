import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { withFormik } from 'formik';
import * as Yup from 'yup';
import CreateListFormModal from './component';
import { createListRequest } from '../../../../../store/myLists/actions';
import constant from '../../../../../helpers/constants';

export class CreateListFormModalContainer extends Component {
  componentDidUpdate() {
    const { status, hideModal, handleReset } = this.props;
    if (status === constant.SUCCESS) {
      hideModal();
      handleReset();
    }
  }

  handleFormCancel = () => {
    const { handleReset, hideModal } = this.props;
    handleReset();
    hideModal();
  };

  handleFormSubmit = () => {
    const { handleSubmit } = this.props;
    handleSubmit();
  };

  render() {
    return (
      <CreateListFormModal
        {...this.props}
        handleFormCancel={this.handleFormCancel}
        handleFormSubmit={this.handleFormSubmit}
      />
    );
  }
}

CreateListFormModalContainer.defaultProps = {
  status: constant.WAITING,
};

CreateListFormModalContainer.propTypes = {
  hideModal: PropTypes.func,
  handleReset: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  isSubmitting: PropTypes.bool.isRequired,
  status: PropTypes.string,
};

export const mapPropsToValues = () => ({
  name: '',
  description: '',
});

export const handleSubmit = (values, formikBag) => {
  const { name, description } = values;
  const { createListRequest } = formikBag.props;
  createListRequest(name, description, formikBag);
};

export const validationSchema = Yup.object().shape({
  name: Yup.string()
    .min(3, 'Минимум 3 символа')
    .matches(/^[A-Za-z0-9_-\s]{3,16}$/, 'Только буквы, цифры и тире')
    .required(),
  description: Yup.string()
    .min(3, 'Минимум 3 символа')
    .required(),
});
const mdtp = {
  createListRequest,
};

export default compose(
  connect(
    null,
    mdtp,
  ),
  withFormik({
    mapPropsToValues,
    handleSubmit,
    validationSchema,
  }),
)(CreateListFormModalContainer);
