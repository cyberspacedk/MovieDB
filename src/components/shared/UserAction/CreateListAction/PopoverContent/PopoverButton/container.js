import React from 'react';
import PropTypes from 'prop-types';
import PopoverButton from './component';

const PopoverButtonContainer = ({ movieId, addMovieToList, item }) => {
  const handlerAddMovieToList = () => {
    addMovieToList(item.id, movieId);
    console.log(item.id, movieId);
  };
  return (
    <PopoverButton item={item} handlerAddMovieToList={handlerAddMovieToList} />
  );
};

PopoverButtonContainer.propTypes = {
  movieId: PropTypes.number,
  item: PropTypes.arrayOf(PropTypes.object),
  addMovieToList: PropTypes.func,
};
export default PopoverButtonContainer;
