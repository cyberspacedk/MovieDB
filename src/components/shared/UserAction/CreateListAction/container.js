import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  getCreatedListRequest,
  addMovieToListRequest,
} from '../../../../store/myLists/actions';
import { getMyList } from '../../../../store/myLists/selectors';
import CreateListAction from './component';

class CreateListActionContainer extends Component {
  state = {
    visiblePop: false,
    visibleMod: false,
  };

  componentDidMount() {
    // eslint-disable-next-line react/destructuring-assignment
    this.props.getCreatedListRequest();
  }

  showPopover = visible => this.setState({ visiblePop: visible });

  hidePopover = () => this.setState({ visiblePop: false });

  showModal = () => this.setState({ visibleMod: true });

  hideModal = () => this.setState({ visibleMod: false });

  addMovieToList = (listId, movieId) => {
    this.props.addMovieToListRequest(listId, movieId);
    this.hidePopover();
  };

  render() {
    const { visiblePop, visibleMod } = this.state;

    return (
      <CreateListAction
        visiblePop={visiblePop}
        visibleMod={visibleMod}
        showPopover={this.showPopover}
        hidePopover={this.hidePopover}
        showModal={this.showModal}
        hideModal={this.hideModal}
        addMovieToList={this.addMovieToList}
        {...this.props}
      />
    );
  }
}

CreateListActionContainer.propTypes = {
  getCreatedListRequest: PropTypes.func.isRequired,
  addMovieToListRequest: PropTypes.func.isRequired,
  myLists: PropTypes.arrayOf(PropTypes.object).isRequired,
};

const mstp = store => ({
  myLists: getMyList(store),
});

const mdtp = {
  getCreatedListRequest,
  addMovieToListRequest,
};

export default connect(
  mstp,
  mdtp,
)(CreateListActionContainer);
