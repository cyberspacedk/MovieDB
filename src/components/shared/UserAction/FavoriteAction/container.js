import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import FavoriteAction from './component';
import { operationsFavoritesRequest } from '../../../../store/favorites/actions';

class FavoriteActionContainer extends Component {
  state = {
    favoriteStatus: false,
  };

  handleFavorite = () => {
    const { operationsFavoritesRequest, movieId } = this.props;
    this.setState(prevState => ({ favoriteStatus: !prevState.favoriteStatus }));
    operationsFavoritesRequest(movieId, true);
  };

  render() {
    const { favoriteStatus } = this.state;
    return (
      <FavoriteAction
        handleFavorite={this.handleFavorite}
        favoriteStatus={favoriteStatus}
      />
    );
  }
}

FavoriteActionContainer.propTypes = {
  operationsFavoritesRequest: PropTypes.func.isRequired,
  movieId: PropTypes.number.isRequired,
};

const mdtp = {
  operationsFavoritesRequest,
};

export default connect(
  null,
  mdtp,
)(FavoriteActionContainer);
