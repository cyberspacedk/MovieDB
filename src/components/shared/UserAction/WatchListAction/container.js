/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import WatchListAction from './component';
import { operationsWatchListRequest } from '../../../../store/watchList/actions';

class WatchListActionContainer extends Component {
  state = {
    watchListStatus: false,
  };

  handleWatchList = () => {
    this.setState(prevState => ({
      watchListStatus: !prevState.watchListStatus,
    }));
    this.props.operationsWatchListRequest(this.props.movieId, true);
  };

  render() {
    const { watchListStatus } = this.state;
    return (
      <WatchListAction
        handleWatchList={this.handleWatchList}
        watchListStatus={watchListStatus}
      />
    );
  }
}

const mdtp = {
  operationsWatchListRequest,
};

export default connect(
  null,
  mdtp,
)(WatchListActionContainer);
