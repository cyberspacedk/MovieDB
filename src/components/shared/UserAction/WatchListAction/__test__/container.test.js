import React from 'react';
import { shallow } from 'enzyme';
import { WatchListActionContainer } from '../container';

describe('WatchListActionContainer ', () => {
  const props = {
    operationsWatchListRequest: jest.fn(),
    movieId: 777,
  };

  const wrapper = shallow(<WatchListActionContainer {...props} />);

  it('Should match its snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  // проверить вызов ф-ции handleWatchList
});
