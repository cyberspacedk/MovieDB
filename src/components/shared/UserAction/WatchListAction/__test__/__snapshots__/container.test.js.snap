// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WatchListActionContainer  Should match its snapshot 1`] = `
<WatchListAction
  handleWatchList={[Function]}
  watchListStatus={false}
/>
`;
